### PR TITLE
Samba custom resources

### PR DESCRIPTION
**Pull request by damacus**
samba_server #36 :
- [x] Installs samba server-related packages
- [x] Managing users as passed into the resource
- [x] Rendering the smb.conf file (with the `samba_share` accumulation)
- [x] Managing services

samba_share #35 :
- [x] Creates the directory for the share
- [x] Appending the configuration to an accumulator for /etc/samba/smb.conf

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +19,7 @@ Installs and configures Samba daemon. Uses Chef Server for data bag to build con
 
 ### Chef
 
-- Chef 11+
+- Chef 12.5+
```
Comment (tas50) [inline README.md]: You can use compat_resource to get 12.1 - 12.4 back

Comment (tas50) [inline README.md]: Or not. It's been a year since 12.6 came out

**Comment by tas50** [comment]
You can disable FC016 with a .foodcritic file since it doesn't know how to deal with custom resources

File: README.md [hunk 2]
```diff
@@ -2,7 +2,13 @@
 
 [![Build Status](https://travis-ci.org/sous-chefs/samba.svg?branch=master)](https://travis-ci.org/sous-chefs/samba) [![Cookbook Version](https://img.shields.io/cookbook/v/samba.svg)](https://supermarket.chef.io/cookbooks/samba)
 
-Installs and configures Samba daemon. Uses Chef Server for data bag to build configuration file shares. Includes a resource for adding / removing Samba users.
+Installs and configures Samba daemon. Uses Chef Server for data bag to build configuration file shares.
+
+# Resources
```
Comment (tas50) [inline README.md]: This is going to format really oddly if you collapse markdown and seems to duplicate the resources section below

**Comment by tas50** [comment]
You can probably delete a good chunk of the chefspecs for now

File: metadata.rb [hunk 3]
```diff
@@ -9,10 +9,12 @@
 recipe 'samba::client', 'Installs smbclient package'
 recipe 'samba::server', 'Installs samba server packages and configures smb.conf'
 
-%w( debian ubuntu centos fedora redhat scientific amazon oracle ).each do |os|
```
Comment (tas50) [inline metadata.rb]: scientifc and oracle just come along for free with RHEL. We actually have a bento box for oracle too if you want to test it.

**Comment by damacus** [comment]
Looks like all the conflicts are files I've gone ahead and deleted.

**Comment by damacus** [comment]
The current issue is nmb doesn't start on systemd as it can't find an available local interface in Vagrant.

File: test/fixtures/cookbooks/test/recipes/client.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
-execute 'apt-get update' if platform_family?('debian')
+apt_update 'update' if platform_family?('debian')
```
Comment (tas50) [inline test/fixtures/cookbooks/test/recipes/client.rb]: the platform check can go away with new chef clients. I'm fairly certain we've fixed that finally

File: resources/server.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  end
+}
+property :max_log_size, String, default: '5000' # 5M
+property :options, [String, nil], default: nil
+property :enable_users_search, [TrueClass, FalseClass], default: true
+property :shares, [Hash, nil], default: nil
+property :config_file, String, default: '/etc/samba/smb.conf'
+property :samba_services, Array, default: lazy {
+  case node['platform']
+  when 'rhel', 'fedora', 'centos'
```
Comment (tas50) [inline resources/server.rb]: This should probably just be platform_family so you catch oracle

File: resources/share.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+property :share_name, String, name_property: true
+property :comment, String
+property :path, String, required: true
+property :guest_ok, String, default: 'no', equal_to: %w(yes no)
+property :printable, String, default: 'no', equal_to: %w(yes no)
+property :write_list, Array, required: true # e.g. %w(jtimberman damacus)
+property :create_mask, String, required: true # e.g. 0644
+property :directory_mask, String, required: true # e.g. 0700
+property :config_file, String, default: lazy {
+  if node['platform_family'] == 'smartos'
```
Comment (tas50) [inline resources/share.rb]: We should get smartos in the kitchen config if we're trying to support it here

File: metadata.rb [hunk 7]
```diff
@@ -3,16 +3,15 @@
...
 
-source_url 'https://github.com/sous-chefs/samba' if respond_to?(:source_url)
-issues_url 'https://github.com/sous-chefs/samba/issues' if respond_to?(:issues_url)
-chef_version '>= 11.0' if respond_to?(:chef_version)
+source_url 'https://github.com/sous-chefs/samba'
+issues_url 'https://github.com/sous-chefs/samba/issues'
+chef_version '>= 12.1' if respond_to?(:chef_version)
+
+depends 'compat_resource', '>= 12.16.3'
+depends 'yum'
```
Comment (tas50) [inline metadata.rb]: Do we actually need this anymore? Seems like we can just move this to berksfile for the testing stuff

Comment (damacus) [inline metadata.rb]: Done. And Fedora still works.

**Pull request merged**